From 187fa0e563a78d2218e2543c83ac61be97bd57e3 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:16:46 -0400
Subject: [net] netfilter: nft_ct: remove family from struct nft_ct

Message-id: <20141021161646.22888.16874.stgit@localhost.localdomain>
Patchwork-id: 97921
O-Subject: [RHEL7 net PATCH 025/124] netfilter: nft_ct: remove family from struct nft_ct
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: d46f2cd2601d01d54fd556395483fb4032155c3b

 commit d46f2cd2601d01d54fd556395483fb4032155c3b
 Author: Patrick McHardy <kaber@trash.net>
 Date:   Fri Mar 7 19:08:32 2014 +0100

    netfilter: nft_ct: remove family from struct nft_ct

    Since we have the context available during destruction again, we can
    remove the family from the private structure.

    Signed-off-by: Patrick McHardy <kaber@trash.net>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nft_ct.c b/net/netfilter/nft_ct.c
index 65a2c7b..bd0d41e 100644
--- a/net/netfilter/nft_ct.c
+++ b/net/netfilter/nft_ct.c
@@ -24,11 +24,10 @@
 struct nft_ct {
  enum nft_ct_keys key:8;
  enum ip_conntrack_dir dir:8;
- union{
+ union {
   enum nft_registers dreg:8;
   enum nft_registers sreg:8;
  };
- uint8_t   family;
 };
 
 static void nft_ct_get_eval(const struct nft_expr *expr,
@@ -316,17 +315,13 @@ static int nft_ct_init(const struct nft_ctx *ctx,
  if (err < 0)
   return err;
 
- priv->family = ctx->afi->family;
-
  return 0;
 }
 
 static void nft_ct_destroy(const struct nft_ctx *ctx,
       const struct nft_expr *expr)
 {
- struct nft_ct *priv = nft_expr_priv(expr);
-
- nft_ct_l3proto_module_put(priv->family);
+ nft_ct_l3proto_module_put(ctx->afi->family);
 }
 
 static int nft_ct_get_dump(struct sk_buff *skb, const struct nft_expr *expr)
-- 
1.7.1